google/cloud/aiplatform/models.py

```diff
@@ -2399,24 +3110,36 @@ def _deploy(
                 Optional. Target Accelerator Duty Cycle.
                 Must also set accelerator_type and accelerator_count if specified.
                 A default value of 60 will be used if not specified.
+
         Returns:
-            endpoint ("Endpoint"):
+            endpoint (Union[Endpoint, PrivateEndpoint]):
                 Endpoint with the deployed model.
         """
 
         if endpoint is None:
             display_name = self.display_name[:118] + "_endpoint"
-            endpoint = Endpoint.create(
-                display_name=display_name,
-                project=self.project,
-                location=self.location,
-                credentials=self.credentials,
-                encryption_spec_key_name=encryption_spec_key_name,
-            )
+
+            if not isinstance(endpoint, PrivateEndpoint):
```

Comment:
endpoint is always `None` based on the if statement above, shouldn't this be `if not network`? If so, preference to add unit test cases to catch this issue.